File: dummy-prior-engine/src/main/java/org/eu/multiply/priors/dummy/DummyPriorEngine.java
package org.eu.multiply.priors.dummy;

import org.eu.multiply.auxdata.dummy.AuxData;

/**
 * @author Tonio Fincke
 */
public class DummyPriorEngine implements PriorEngine {

    public Prior createPrior(AuxData auxData) {
        return new DummyPrior();
    }

    public Prior getPrior(String name) {
        return new DummyPrior();
    }

    public void savePrior(Prior prior) {
        // do nothing
    }

}

File: dummy-prior-engine-python/dummy_prior_engine_python/dummy_prior_engine.py
class dummy_prior_engine:

    def create_prior(self, aux_data):
        print 'I create a prior'
        return []

    def save_prior(self, prior):
        print 'I save a prior'

    def get_prior(self, prior_id):
        print 'I return a prior'
        return []